### DISCUSSION TITLE
How to match last of a range of statements in a block?

### DISCUSSION THREAD

**Discussion by madskjeldgaard** [tree-sitter/tree-sitter#899] [Q&A] [answered]
Hello all

I am writing my first tree-sitter grammar and it's going fairly well. I am writing it for the SuperCollider programming language (yay!). I am running into a substantial problem though that I am not sure how to solve. In SuperCollider, a code block is defined by a parenthesis with a range of statements in them. All statements end with a semicolon. The exception to this rule is the last statement where you may omit the semicolon (which is because the last statement is treated as a return statement).  

Does anyone have any tips on how to achieve a match on that last statement which is then allowed to not have a semicolon?

Examples:
```
(
1+2;
10+20;
2
)
```

Thanks for your help and for building tree-sitter ! 

best regards

**Comment by razzeee** [comment, marked as answer]
```
wrapper: ($) => seq("(", $.body, ")");
body: ($) => seq(repeat(seq($.statement, ";")), $.statement);
statement ($) => // your statement handling
```
1. please choose better names
2. it's possible I misunderstood you
3. this assumes, that only a return is okay too

**Comment by madskjeldgaard** [comment]
Thank you! I think this is exactly what I wanted. I'll try it out!
